src/uu/runcon/src/runcon.rs

```diff
@@ -0,0 +1,453 @@
+        range: Option<OsString>,
+
+        // `command` can be `None`, in which case we're dealing with this syntax:
+        // runcon [-c] [-u USER] [-r ROLE] [-t TYPE] [-l RANGE]
+        //
+        // This syntax is undocumented, but it is accepted by the GNU implementation,
+        // so we do the same for compatibility.
+        command: Option<OsString>,
+    },
+}
+
+#[derive(Debug)]
+struct Options {
+    mode: CommandLineMode,
+    arguments: Vec<OsString>,
+}
+
+fn parse_command_line(config: App, args: impl uucore::Args) -> Result<Options> {
+    let matches = config.get_matches_from_safe(args)?;
+
+    let compute_transition_context = matches.is_present(options::COMPUTE);
+
+    let mut args = matches
+        .values_of_os("ARG")
+        .unwrap_or_default()
+        .map(OsString::from);
+
+    if compute_transition_context
+        || matches.is_present(options::USER)
+        || matches.is_present(options::ROLE)
+        || matches.is_present(options::TYPE)
+        || matches.is_present(options::RANGE)
+    {
+        // runcon [-c] [-u USER] [-r ROLE] [-t TYPE] [-l RANGE] [COMMAND [args]]
+
+        let mode = CommandLineMode::CustomContext {
+            compute_transition_context,
+            user: matches.value_of_os(options::USER).map(Into::into),
+            role: matches.value_of_os(options::ROLE).map(Into::into),
+            the_type: matches.value_of_os(options::TYPE).map(Into::into),
+            range: matches.value_of_os(options::RANGE).map(Into::into),
+            command: args.next(),
+        };
+
+        Ok(Options {
+            mode,
+            arguments: args.collect(),
+        })
+    } else if let Some(context) = args.next() {
+        // runcon CONTEXT COMMAND [args]
+
+        args.next()
+            .ok_or(Error::MissingCommand)
+            .map(move |command| Options {
+                mode: CommandLineMode::PlainContext { context, command },
+                arguments: args.collect(),
+            })
+    } else {
+        // runcon
+
+        Ok(Options {
+            mode: CommandLineMode::Print,
+            arguments: Vec::default(),
+        })
+    }
+}
+
+fn print_current_context() -> Result<()> {
+    let op = "Getting security context of the current process";
+    let context = SecurityContext::current(false).map_err(|r| Error::from_selinux(op, r))?;
+
+    let context = context
+        .to_c_string()
+        .map_err(|r| Error::from_selinux(op, r))?;
+
+    if let Some(context) = context {
+        let context = context.as_ref().to_str()?;
+        println!("{}", context);
+    } else {
+        println!();
+    }
+    Ok(())
+}
+
+fn set_next_exec_context(context: OpaqueSecurityContext) -> Result<()> {
+    let c_context = context
+        .to_c_string()
+        .map_err(|r| Error::from_selinux("Creating new context", r))?;
+
+    let sc = SecurityContext::from_c_str(&c_context, false);
+
+    if sc.check() != Some(true) {
+        use std::os::unix::ffi::OsStrExt;
```

Comment:
Because I always keep platform-specific `use` statements in the function using them. Along with marking the function `#[cfg(unix)]`, this allows implementing the same function for different platforms as needed. In this particular case, it's out of habit.